backend/src/main/java/ai/verta/modeldb/batchProcess/CollaboratorResourceMigration.java

```diff
@@ -152,6 +153,84 @@ private void migrateProjects() {
+
+    int lowerBound = 0;
+    final int pagesize = this.paginationSize;
+    LOGGER.debug("Total Repositories {}", count);
+
+    while (lowerBound < count) {
+
+      try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
+        CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+
+        CriteriaQuery<RepositoryEntity> criteriaQuery =
+            criteriaBuilder.createQuery(RepositoryEntity.class);
+        Root<RepositoryEntity> root = criteriaQuery.from(RepositoryEntity.class);
+
+        CriteriaQuery<RepositoryEntity> selectQuery =
+            criteriaQuery.select(root).orderBy(criteriaBuilder.asc(root.get("id")));
+
+        TypedQuery<RepositoryEntity> typedQuery = session.createQuery(selectQuery);
+
+        typedQuery.setFirstResult(lowerBound);
+        typedQuery.setMaxResults(pagesize);
+        List<RepositoryEntity> repositoryEntities = typedQuery.getResultList();
+
+        if (repositoryEntities.size() > 0) {
+          Set<String> userIds = new HashSet<>();
+          for (RepositoryEntity repositoryEntity : repositoryEntities) {
+            userIds.add(repositoryEntity.getOwner());
+          }
+          LOGGER.debug("Repository userId list : " + userIds);
+
+          // Fetch the repository owners userInfo
+          Map<String, UserInfo> userInfoMap =
+              authService.getUserInfoFromAuthServer(userIds, null, null);
+          for (RepositoryEntity repository : repositoryEntities) {
+            try {
+              WorkspaceDTO workspaceDTO =
+                  roleService.getWorkspaceDTOByWorkspaceId(
+                      userInfoMap.get(repository.getOwner()),
+                      repository.getWorkspace_id(),
+                      repository.getWorkspace_type());
+              // if repositoryVisibility is not equals to ResourceVisibility.ORG_SCOPED_PUBLIC then
+              // ignore the CollaboratorType
+              roleService.createWorkspacePermissions(
+                  workspaceDTO.getWorkspaceName(),
+                  String.valueOf(repository.getId()),
+                  Optional.of(Long.parseLong(repository.getOwner())),
+                  repository.isDataset()
+                      ? ModelDBServiceResourceTypes.DATASET
+                      : ModelDBServiceResourceTypes.REPOSITORY,
+                  CollaboratorTypeEnum.CollaboratorType.READ_ONLY,
+                  getResourceVisibility(
+                      Optional.ofNullable(
+                          WorkspaceTypeEnum.WorkspaceType.forNumber(
+                              repository.getWorkspace_type())),
+                      VisibilityEnum.Visibility.forNumber(repository.getRepository_visibility())));
+            } catch (Exception ex) {
+              LOGGER.warn("CollaboratorResourceMigration Exception: {}", ex.getMessage());
+            }
+          }
+        } else {
+          LOGGER.debug("Total repositories count 0");
+        }
```

Comment:
Same comments about exception as the other PR.

Reply:
Throw an exception to run the service noisy but other core changes are already done on based PR so it will be added while we merge this.